### PR TITLE
Add tests loading hazard curves, hazard maps and uhs from npz

### PR DESCRIPTION
**Pull request by ptormene**
Fixes https://github.com/gem/oq-irmt-qgis/issues/172
These tests load the dialog that is used to import data from npz files produced by the oq-engine. The files were obtained running the hazard demo `AreaSourceClassicalPSHA`.
The dialog is opened and "accepted" without changing any of its elements. Once it is accepted, the corresponding layer is built (i.e.: a hazard map, hazard curves and uniform hazard spectra).
I am also selecting some features from uhs and from hazard curves, driving also the Data Viewer.
Finally, the `export as csv` button is triggered and the exported file is compared to a reference (both for uhs and for hazard curves).

In order to make tests pass, I had to make a little hack, as described in https://github.com/gem/oq-irmt-qgis/issues/177
Apart from that, things are working now and I think this PR is ready to be merged.

### CODE REVIEW

File: svir/dialogs/drive_oq_engine_server_dialog.py [hunk 1]
```diff
@@ -132,7 +132,7 @@ def login(self):
     def is_lockdown(self):
         # try retrieving the list of calculations and see if the server
         # redirects you to the login page
-        # NOTE: we might call something quicker than this, if available 
+        # NOTE: we might call something quicker than this, if available
         calc_list_url = "%s/v1/calc/list?relevant=true" % self.hostname
```
Comment (micheles) [inline svir/dialogs/drive_oq_engine_server_dialog.py]: You may try `""%s/engine_version" % self.hostname` here

**Comment by micheles** [comment]
LGTM

**Comment by ptormene** [comment]
I re-added the WIP, because tests are not reliable yet.

File: svir/test/test_load_npz_as_layer.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        IFACE.newProject()
+        curr_dir_name = os.path.dirname(__file__)
+        self.data_dir_name = os.path.join(
+            curr_dir_name, 'data', 'hazard')
+        mock_action = QAction(IFACE.mainWindow())
+        self.viewer_dock = ViewerDock(IFACE, mock_action)
+
+    def tearDown(self):
+        IFACE.newProject()
+
```
Comment (micheles) [inline svir/test/test_load_npz_as_layer.py]: Please add a comment that this operation clean up all the existing layers

File: svir/test/test_load_npz_as_layer.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        self._change_selection()
+        # test exporting the current selection to csv
+        self._test_export('uniform_hazard_spectra.csv')
+
+    def _test_export(self, expected_file_name):
+        _, exported_file_path = tempfile.mkstemp(suffix=".csv")
+        layers = CANVAS.layers()
+        layer = layers[-1]
+        layer.select([1, 2])
+        # probably we have the wrong layer selected (uhs produce many layers)
```
Comment (micheles) [inline svir/test/test_load_npz_as_layer.py]: what is selecting here?

**Comment by micheles** [comment]
LGTM apart from the minor comments

**Pull request merged**